integrationtests/src/test/java/org/neo4j/causalclustering/scenarios/ListQueriesProcedureInClusterIT.java

```diff
@@ -0,0 +1,172 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.scenarios;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.neo4j.causalclustering.core.CoreGraphDatabase;
+import org.neo4j.causalclustering.discovery.Cluster;
+import org.neo4j.causalclustering.readreplica.ReadReplicaGraphDatabase;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.impl.factory.GraphDatabaseFacade;
+import org.neo4j.test.causalclustering.ClusterRule;
+import org.neo4j.test.rule.VerboseTimeout;
+import org.neo4j.test.rule.concurrent.ThreadingRule;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+
+public class ListQueriesProcedureInClusterIT {
+    private final ClusterRule clusterRule =
+            new ClusterRule(getClass()).withNumberOfCoreMembers(3).withNumberOfReadReplicas(1);
+    private final VerboseTimeout timeout = VerboseTimeout.builder()
+            .withTimeout(1000, TimeUnit.SECONDS)
+            .build();
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(clusterRule).around(timeout);
+
+    private Cluster cluster;
+
+    @Rule
+    public final ThreadingRule threads = new ThreadingRule();
+
+    @Before
+    public void setup() throws Exception {
+        cluster = clusterRule.startCluster();
+    }
+
+    @Test
+    public void shouldListQueriesOnCoreAndReplica() throws Exception {
+        // given
+        String CORE_QUERY = "MATCH (n) SET n.number = n.number - 1";
+        String REPLICA_QUERY = "MATCH (n) RETURN n";
+        BiConsumer<Transaction, Node> lock = (transaction, entity) -> transaction.acquireWriteLock(entity);
+        CountDownLatch resourceLocked = new CountDownLatch(1);
+        CountDownLatch listQueriesLatch = new CountDownLatch(1);
+
+        cluster.coreTx((CoreGraphDatabase leaderDb, Transaction tx) ->
+        {
+            try {
+
+                //Given
+                ReadReplicaGraphDatabase replicaDb = cluster.findAnyReadReplica().database();
+                Node node = leaderDb.createNode();
+                tx.success();
```

Comment:
Since the transaction doesn't actually commit until `tx.close()` is invoked, the transaction has not committed at this point.

This means that the rest of this method relies on racy behaviour. It is quite possible that the execution of `CORE_QUERY` (and `REPLICA_QUERY`) will match no nodes at all, in which case the `CORE_QUERY` will not block.

The thread that locks the node will actually succeed, even though it shouldn't, this is for two reasons:
1. A transaction that creates a node does not lock it, since it should not be visible to any other transaction.
2. When locking a node, we don't actually check to see that it exists / is visible to that transaction.

I really don't think we should rely on such accidental behaviour.